src/replication/ng/dao/policy.go

```diff
@@ -0,0 +1,115 @@
+package dao
+
+import (
+	"errors"
+	"time"
+
+	"github.com/astaxie/beego/orm"
+	common_dao "github.com/goharbor/harbor/src/common/dao"
+	"github.com/goharbor/harbor/src/replication/ng/dao/model"
+)
+
+// AddRepPolicy insert new policy to DB.
+func AddRepPolicy(policy model.RepPolicy) (int64, error) {
+	o := common_dao.GetOrmer()
+	now := time.Now()
+	policy.CreationTime = now
+	policy.UpdateTime = now
+
+	return o.Insert(policy)
+}
+
+func filteredRepPolicyQuerySeter(name, namespace string) orm.QuerySeter {
+	var qs = common_dao.GetOrmer().QueryTable(new(model.RepPolicy))
+
+	var cond = orm.NewCondition()
+	// Pull-based polices in the special namespace.
+	// If dest_registry_id equal 0, dest_namespace is the namespace of Harbor.
+	pullBasedCond := cond.And("dest_registry_id", 0).And("dest_namespace", namespace)
+	// Push-based polices in the special namespace.
+	// If src_registry_id equal 0, src_namespaces only have single namespace of Harbor.
+	pushBasedCond := cond.And("src_registry_id", 0).And("src_namespaces", namespace)
```

Comment:
Source namespaces should be an array, seems the logic here is not correct

Reply:
The namespaces is array in the manager's model, but it is serialize to string stored in database, and split with comma. In push-based mode, only one namespace in the namespaces array.

Reply:
The source namespaces may have more than one element in both pull and push mode.

Reply:
Replicate multi-namespaces from harbor to another registry in one policy?I don't think it is a good idea.

Reply:
By the sync meeting, we just filter name now.